core/src/main/scala/kafka/log/Log.scala

```diff
@@ -292,6 +305,20 @@ class Log(val dir: File,
    */
   def numberOfSegments: Int = segments.size
 
+
+  /**
+   * Returns compiled information about segments
+   */
+  def segmentsInfo: util.ArrayList[String] = {
+    val list = new util.ArrayList[String]()
+    logSegments.foreach(
+      seg => list.add("baseOffset=%s, created=%s, logSize=%s, indexSize=%s".format(
+        seg.baseOffset, seg.created, seg.size, seg.index.sizeInBytes()
+      ))
+    )
+    list
```

Comment:
This can be implemented more nicely by using `map`, eg

```scala
logSegments.toSeq.map(seg => s"baseOffset=${seg.baseOffset}, created=${seg.created}, logSize=${seg.size}, indexSize=$[seq.index.sizeInBytes}").asJava
```